applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -220,7 +220,215 @@ def GetShapeGradient(self):
         return gradient
 
 # ==============================================================================
+class AdjointResponseFunction(ResponseFunctionBase):
+    """Linear static adjoint response function.
+    - runs the primal analysis (writes the primal results to an .h5 file)
+    - reads the primal results from the .h5 file into the adjoint model part
+    - uses primal results to calculate value
+    - uses primal results to calculate gradient by running the adjoint analysis
 
+    Attributes
+    ----------
+    primal_analysis : Primal analysis object of the response function
+    adjoint_analysis : Adjoint analysis object of the response function
+    response_function_utility: Cpp utilities object doing the actual computation of response value and gradient.
+    """
+    def __init__(self, identifier, project_parameters, model_part = None):
+        self.identifier = identifier
+
+        model = Model()
```

Comment:
Who is owning the model?

Reply:
The analysis owns it, but i see now, that this is not a clean solution. 
This will be updated, once the shape optimization is adapted to use the Model instead of model parts.

Reply:
For now: The model is stored inside the analysis object and is owned by it.

Reply:
ok